File: LeetCode/14.Longest_Common_Prefix/14.Longest_Common_Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        // set strs[0] as Prefix, then compare the next
        // if common prefix is "", return it
        // if not, continue compare the next, unitl the last one

        String prefix = strs[0];
        for (int i = 1; i < strs.length; i++) {
            String cur = strs[i];
            while (cur.indexOf(prefix) != 0) {
                prefix = prefix.substring(0, prefix.length() - 1);
                if (prefix.isEmpty()) {
                    return "";
                }
            }
        }

        return prefix;
    }
}

File: LeetCode/14.Longest_Common_Prefix/14.Longest_Common_Prefix.swift
class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        guard strs.count >= 2 else { return strs[0] }

        var prefix = strs[0]
        for i in 1 ..< strs.count {
            let curr = strs[i]
            while !curr.hasPrefix(prefix) {
                if prefix.isEmpty { return "" }
                prefix.removeLast()
            }
            if prefix.isEmpty { return "" }
        }

        return prefix
    }
}
